packages/network-clients/src/networkClient.ts

```diff
@@ -61,6 +62,20 @@ export class NetworkClient {
   //   return JSON.parse(metadataStr);
   // }
 
+  public async maxUnstakeAmount(address: string): Promise<BigNumber> {
+    const leverageLimit = await this._sdk.staking.indexerLeverageLimit();
+    const minStakingAmount = await this._sdk.indexerRegistry.minimumStakingAmount();
+
+    const { totalStake }  = await this._gqlClient.getIndexer(address);
+    const { amount } = await this._gqlClient.getDelegation(address, address);
+
+    const totalStakingAmountAfter = BigNumber.from(totalStake?.after ?? 0); 
+    const ownStakeAfter = BigNumber.from(amount?.valueAfter?.value ?? 0);
+
+    const maxUnstakeAmount = min(ownStakeAfter.sub(minStakingAmount), ownStakeAfter.mul(leverageLimit)).sub(totalStakingAmountAfter).div(leverageLimit.sub(1))
```

Comment:
There is an potential issue when `leverageLimit` equals `1`. That means indexer can not get any delegation.

```
if (leverageLimit === 1) return ownStakeAfter.sub(minStakingAmount)

```